src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -2848,4 +2953,128 @@ public function is_development_environment() {
 		return in_array( wp_get_environment_type(), array( 'development', 'local' ), true );
 	}
 
+	/**
+	 * Determines whether to suggest using a persistent object cache.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @return bool Whether to suggest using a persistent object cache.
+	 */
+	public function should_suggest_persistent_object_cache() {
+		global $wpdb;
+
+		if ( is_multisite() ) {
+			return true;
+		}
+
+		/**
+		 * Filters to force suggestion to use a persistent object cache and bypass threshold checks.
+		 *
+		 * @since 6.1.0
+		 *
+		 * @param bool $suggest Whether to suggest using a persistent object cache.
+		 */
+		if ( apply_filters( 'site_status_suggest_persistent_object_cache', false ) ) {
+			return true;
+		}
+
+		/**
+		 * Filters the thresholds used to determine whether to suggest the use of a persistent object cache.
+		 *
+		 * @since 6.1.0
+		 *
+		 * @param array $thresholds The list of threshold names and numbers.
+		 */
+		$thresholds = apply_filters(
+			'site_status_persistent_object_cache_thresholds',
+			array(
+				'alloptions_count' => 500,
+				'alloptions_bytes' => 100000,
+				'comments_count'   => 1000,
+				'options_count'    => 1000,
+				'posts_count'      => 1000,
+				'terms_count'      => 1000,
+				'users_count'      => 1000,
+			)
+		);
+
+		$alloptions = wp_load_alloptions();
+
+		if ( $thresholds['alloptions_count'] < count( $alloptions ) ) {
+			return true;
+		}
+
+		if ( $thresholds['alloptions_bytes'] < strlen( serialize( $alloptions ) ) ) {
+			return true;
+		}
+
+		$table_names = implode( "','", array( $wpdb->comments, $wpdb->options, $wpdb->posts, $wpdb->terms, $wpdb->users ) );
+
+		// With InnoDB the `TABLE_ROWS` are estimates, which are accurate enough and faster to retrieve than individual `COUNT()` queries.
+		$results = $wpdb->get_results(
+			$wpdb->prepare(
+			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
```

Comment:
I'd love for any line that disables coding standard checks to include a comment on _why_ the standards are being ignored, and why that is expected. I couldn't find any guidelines on this, and am checking in with the standards team to make sure we do this the right way :)

Reply:
I removed the second ignore which was not needed. However, this one is needed for 
`AND TABLE_NAME IN ('$table_names')`

@Clorith let me know when you hear from the standards team.

Reply:
@manuelRod: You removed the re-enable comment, which needs to stay.

Reply:
There's no hard guideline on requiring omments per now, but the team agreed that this is something that should be created, so let's move forward with adding a comment on why a rule is being ignored here, be the change we want to see in the world 😃 

As for enabling/disabling of rules, you could also do a single-line-allow by using `// phpcs:ignore <rule name> -- <comment with reason for ignoring the rule right here>` for the single `$wpdb->prepare()` command, that may also be handy to not accidentally forget to re-enable a rule later on.

Reply:
@Clorith The wpdb query is extremely fast even on massive sites, I don't think we need to make this an async method.

Reply:
@manuelRod: Let's add the reason then. This query cannot use interpolation.

Reply:
Ignore comment added.

Reply:
fixed